### ISSUE TITLE
App usage records statistics does not match

### ISSUE THREAD

**Issue by javlon** [aliannejadi/LSApp#1]
I have calculated statistics for app usage records, and it does not match with statistics from readme (and paper). Other statistics (sessions, unique apps, users, mean duration/user) matches with numbers from table  in readme file.

Also I have observed that (train, validation, test) instances from statistics table does not sum up to app usage records statistic (663,954 vs 599,635).

I have tried to cleanse data with several ways, but could not obtain a number even close to one I mentioned above.

Records filtering method mentioned in paper:
> *we considered all repeated app usage records with less than one minute time difference as one
record*. 

I have tried to merge only consecutive app records as in quote and got 692,217 apps usage record. When I merged with ignoring consecutive requirement, I got 346,549 records.

Questions:
1. Why do not (train, validation, test) instances sum up to number of app usage records?
2. What are the cleansing methods should be applied to obtain same numbers as you put in data statistics table?
3. Some users have small number of records (less than 10), they can't be splitted to (train, validation, test) instances. Did you filter (remove) those users?

**Comment by aliannejadi** [comment]
Hi, 
Thank you for your message. I have added a new file called "data_prep" that takes the steps I did on the data to achieve the same data set that I used in the paper. I hope it answers your questions. Regarding your last question, no we did not remove users with few interactions.

Thanks,
Mohammad

**Comment by javlon** [comment]
Hi
Thanks for adding data preparing script. It’s very helpful. 
But I still have one question. Target apps recommendation predicts next app, and number of instances should not exceed number of app usage records. But #instances is almost 10% more than #app usage records. Is it typo in number of instances?

**Comment by aliannejadi** [comment]
Hi
Thank you for your response. Can you please be more specific? Which numbers
in the paper are you referring to? I can check them in a few days.

**Comment by javlon** [comment]
Apologies for not being clear.
Bottom part of *"Table 2. Statistics of LSApp."* contains following information:
|||
|-|-|
|# train instances| 464,903|
|# validation instances |66,300|
|# test instances |132,751|

I assume that it is the size of train, validation and test sets for NeuSA model. Because it has same proportions (70%, 10% and 20%) as you wrote in paper:
> **Data.** For every user, we take the 70% earliest app usage records as training set, 10% next records
as validation, and 20% latest records as test set.

NeuSA solves following problem: to predict next app that the user would want to launch or interact with. It means each app usage record forms one instance. If we split records into train, valiadation and test without intersection then number of train, validation and test instances can't be greater than number of app usage records. But *Table 2* shows that 464,903+66,300+132,751=663,954 **>** 599,635 which is almost 10% higher.

Questions:
* Why number of instances does not sum up to number of app usage?
* If each app usage record does not generate only one instance then how should we form instances?

**Comment by aliannejadi** [comment]
Thanks for clarification, and apologies for the delay (I was on holidays).
This must be a typo in the numbers. Those numbers should sum up to the total number instances. I can look up later what the exact number is.
If you look at the code that I’ve added for processing data, we make some assumptions regarding multiple instances of app usages for the same app and user. Unfortunately, Android app usage records are not very accurate and that’s the best we can do, I think.

Thanks,
Mohammad
